test/services/AcceptStakeRequestService/update.test.ts

```diff
@@ -0,0 +1,218 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+
+import 'mocha';
+
+import BigNumber from 'bignumber.js';
+import * as sinon from 'sinon';
+import StakeRequest from '../../../src/models/StakeRequest';
+import Repositories from '../../../src/repositories/Repositories';
+import {
+  Message,
+  MessageAttributes,
+  MessageDirection,
+  MessageStatus,
+  MessageType,
+} from '../../../src/repositories/MessageRepository';
+import AcceptStakeRequestService from '../../../src/services/AcceptStakeRequestService';
+import assert from '../../test_utils/assert';
+
+const Web3 = require('web3');
+
+interface TestConfigInterface {
+  web3: any;
+  repos: Repositories;
+  stakeRequestWithMessageHashB: StakeRequest;
+  stakeRequestWithNullMessageHashC: StakeRequest;
+  service: AcceptStakeRequestService;
+}
+let config: TestConfigInterface;
+
+describe('StakeRequestRepository::save', (): void => {
+  beforeEach(async (): Promise<void> => {
+    const repos = await Repositories.create();
```

Comment:
I feel repositories should be mocked here. It's not a unit test. Actual logic repositories are also executed.

Reply:
I do not think we should mock everything in a single unit test. In this case, it's in memory database. The border is not clear here. However, using in-memory database seems good to me in this case. Thoughts?

Reply:
Then it means, here it's not unit testing. In other services, we have actually mocked repositories and did unit testing. We should crisply define boundaries and make it consistent. 

We should discuss this with the team 
CC @OpenST/maintainers

Reply:
But we can proceed with this PR.

Reply:
Sure. Let's discuss with the team.